System/IO/Temp.hs

```diff
@@ -185,6 +186,22 @@ createTempDirectory dir template = findTempName
         Left  e | isAlreadyExistsError e -> findTempName
                 | otherwise              -> ioError e
 
+-- | Get a temporary file name, without creating a file.
+--
+-- This may be useful if you have some process wait for a file creation to
+-- happen.
```

Comment:
I don't quite follow what you're getting at here.

To me, "wait for a file creation to happen" would be something like `while ! [ -f "$FILE" ]; do sleep 1; done` — but I don't see how it's related to `createTempFileName`.

Reply:
That is more or less what I wanted to say.

So you can have a process/function/whatever that accepts a filename and then waits for it to appear to do something (via inotify or busy loop or whatever). One could get such a filename via `createTempFileName`.

We can remove this comment if it's confusing, it's probably not useful to try to suggest some use-case.